xc7/toolchain_wrappers/synth

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+MYPATH=`realpath $0`
+MYPATH=`dirname ${MYPATH}`
+
+export TECHMAP_PATH=`realpath ${MYPATH}/../share/prjxray/techmap`
+
+SYNTH_TCL_PATH=`realpath ${MYPATH}/../share/prjxray/synth.tcl`
+CONV_TCL_PATH=`realpath ${MYPATH}/../share/prjxray/conv.tcl`
+
+VERILOG_FILES=()
+XDC_FILES=()
+TOP=top
+
+VERILOGLIST=0
+XDCLIST=0
+TOPNAME=0
+
+for arg in $@; do
+	echo $arg
+	case "$arg" in
+		-t|--top)
+			echo "adding top"
+			VERILOGLIST=0
+			XDCLIST=0
+			TOPNAME=1
+			;;
+		-x|--xdc)
+			VERILOGLIST=0
+			XDCLIST=1
+			TOPNAME=0
+			;;
+		-v|--verilog)
+			VERILOGLIST=1
+			XDCLIST=0
+			TOPNAME=0
+			;;
+		*)
+			if [ $VERILOGLIST -eq 1 ]; then
+				VERILOG_FILES+=($arg)
+			elif [ $XDCLIST -eq 1 ]; then
+				XDC_FILES+=($arg)
+			elif [ $TOPNAME -eq 1 ]; then
+				TOP=$arg
+			else
+				echo "Usage: synth [-t|--top <top module name> -v|--verilog <Verilog files list> [-x|--xdc <XDC files list>]"
+				exit 1
+			fi
+			;;
+	esac
+done
+
+if [ ${#VERILOG_FILES[@]} -eq 0 ]; then
+	echo "Please at least one Verilog file"
+	exit 1
+fi
+
+export INPUT_XDC_FILE=${XDC_FILES[*]}
```

Comment:
Missing export of YOSYS_TECHMAP?  This is why adding a test case is important.

Reply:
the script exports `TECHMAP_PATH` here https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1252/files#diff-060a92a8b914fe0d8c57eef57705f9daR6

this variable is used in the tcl https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1252/files#diff-3a4d31cef2659b1011b54cd37fa3edc0R10

> This is why adding a test case is important.

I agree, I'm adding the tests